packages/block-editor/src/components/block-inspector/index.js

```diff
@@ -70,19 +68,21 @@ const BlockInspector = ( {
 					</PanelBody>
 				</div>
 			) }
-			<div><InspectorControls.Slot /></div>
+			<InspectorControls.Slot bubblesVirtually />
 			<div>
-				<InspectorAdvancedControls.Slot>
-					{ ( fills ) => ! isEmpty( fills ) && (
-						<PanelBody
-							className="editor-block-inspector__advanced block-editor-block-inspector__advanced"
-							title={ __( 'Advanced' ) }
-							initialOpen={ false }
-						>
-							{ fills }
-						</PanelBody>
-					) }
-				</InspectorAdvancedControls.Slot>
+				<__experimentalSlotFillConsumer>
```

Comment:
I'm not totally satisfied that I had to expose the consumer. The problem is, in the portal-based slot implementation, the portal should be rendered and can't support the same `children` as function prop we have in the other implementation (there's no "children" to be passed since it's all handled by React).

Reply:
Hmm, is there a way we could assign the slot's `children` function as a property of the slot (`useSlot`) that can be run from the `Fill` ?

Or something like a `getChildren` function called by the fill?

Reply:
But that's not what we want, the requirement here is that we don't want to show the panel component that is rendered as a parent of a slot, if there's no fill for that particular slot.

Reply:
I kept thinking about this and I don't see a better approach for the moment.

Reply:
Have you considered exposing `InspectorAdvancedControls.Consumer` instead? It could have the following characteristics:
- `hasFills` - boolean value
- `slot` - would be a shortcut for `InspectorAdvancedControls.Slot bubblesVirtually />`

In code:

```js
<InspectorAdvancedControls.Consumer>
    { ( { hasFills, slot } ) =>
        hasFills && (
            <PanelBody { ...panelProps }>
                { slot }
            </PanelBody>
        )
    }
</InspectorAdvancedControls.Consumer>

Reply:
How do you build this `InspectorAdvancedControls.Consumer` ?

Reply:
I still don't know how to exactly solve it :)